R/bnma_analysis.R

```diff
@@ -124,3 +124,97 @@ BaselineRiskDicTable <- function(br_model){
   names(dic_table)[4] <- "Data points"
   return(dic_table)
 }
+
+
+
+#' Puts a relative effects table from bnma into gemtc format
+#'
+#' @param median_ci_table Output from bnma::relative.effects.table(, summary_stat = "ci")
+#' @return A relative effects table in the same format as from gemtc.
+BaselineRiskRelativeEffectsTable <- function(median_ci_table){
+  #Entries in the input table are in the form "[lower_ci,median,upper_ci]" (no spaces)
+  
+  #The dimensions of the (square) table
+  dim_median <- nrow(median_ci_table)
+  #Create matrices to store the lower_ci, median and upper_ci separately
+  lower_ci <- matrix(nrow = dim_median, ncol = dim_median)
+  median_br <- matrix(nrow = dim_median, ncol = dim_median)
+  upper_ci <- matrix(nrow = dim_median, ncol = dim_median)
+  
+  #Extract the lower_ci
+  for(row in 1:dim_median){
+    for(col in 1:dim_median){
+      lower_ci[row, col] <- round(
+        as.numeric(
+          substr(x = median_ci_table[row, col],
+                 start = 2,
+                 stop = as.vector(
+                   gregexpr(pattern = ",",
+                            text = median_ci_table[row, col])[[1]]
+                   )[1] - 1
+                 )
```

Comment:
This chunk to extract the first number in the string can be simplified to:
```
median_br[row, col] <- median_ci_table[row, col] %>%
  stringr::str_extract(pattern = "([-0-9\\.]+)") %>%
  as.numeric() %>%
  round(digits = 2)
```
(My best attempt to make it tidyverse-style)

I think that's more readable than using `gregexpr` to get indices, though that does work.

Reply:
I have used str_extract_all as I agree it's more readable. I've also put the three extractions in the same for-loop. But I haven't used the pipe operator because I don't like it.

Reply:
Great idea to combine the loops. I hadn't noticed that.
One extra improvement can therefore be to extract the numbers once, then assign them, rather than running the same regex 3 times:
```
interval <- round(
    as.numeric(
        stringr::str_extract_all(
            string = median_ci_table[row, col],
            pattern = "[-0-9\\.]+"
        )[[1]]
    ), digits = 2
)
lower_ci[row, col] <- interval[1]
median_br[row, col] <- interval[2]
upper_ci[row, col] <- interval[3]
```